test/extended/util/volumes.go

```diff
@@ -0,0 +1,219 @@
+			e2e.Logf("Error creating pv subdir %s: %v\n", rootDir, err)
+			return volumes, err
+		}
+		e2e.Logf("Created pv subdir %s\n", dir)
+		if _, err = exec.LookPath("chcon"); err == nil {
+			e2e.Logf("Found chcon in path\n")
+			//err := exec.Command("chcon", "-t", "container_file_t", dir).Run()
+			out, err := exec.Command("chcon", "-t", "svirt_sandbox_file_t", dir).CombinedOutput()
+			if err != nil {
+				e2e.Logf("Error running chcon on %s, %s, %v\n", dir, string(out), err)
+				return volumes, err
+			}
+			e2e.Logf("Ran chcon on %s\n", dir)
+		}
+		if err != nil {
+			e2e.Logf("Error finding chcon in path: %v\n", err)
+			return volumes, err
+		}
+		if err = os.Chmod(dir, 0777); err != nil {
+			e2e.Logf("Error running chmod on %s, %v\n", dir, err)
+			return volumes, err
+		}
+		e2e.Logf("Ran chmod on %s\n", dir)
+		pv, err := c.Create(CreatePersistentVolume(fmt.Sprintf("%s%s-%0.4d", pvPrefix, prefix, i), capacity, dir))
+		if err != nil {
+			e2e.Logf("Error defining PV %v\n", err)
+			return volumes, err
+		}
+		e2e.Logf("Created PVs\n")
+		volumes = append(volumes, pv)
+	}
+	return volumes, err
+}
+
+// RemoveHostPathVolumes removes all persistent volumes created by SetupHostPathVolumes
+func RemoveHostPathVolumes(oc *CLI) error {
+	c := oc.AdminKubeClient().Core().PersistentVolumes()
+	prefix := oc.Namespace()
+
+	pvs, err := c.List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	prefix = fmt.Sprintf("%s%s-", pvPrefix, prefix)
+	for _, pv := range pvs.Items {
+		if !strings.HasPrefix(pv.Name, prefix) {
+			continue
+		}
+
+		pvInfo, err := c.Get(pv.Name, metav1.GetOptions{})
+		if err != nil {
+			e2e.Logf("WARNING: couldn't get meta info for PV %s: %v\n", pv.Name, err)
+			continue
+		}
+
+		if err = c.Delete(pv.Name, nil); err != nil {
+			e2e.Logf("WARNING: couldn't remove PV %s: %v\n", pv.Name, err)
+			continue
+		}
+
+		volumeDir := pvInfo.Spec.HostPath.Path
+		if err = os.RemoveAll(volumeDir); err != nil {
+			e2e.Logf("WARNING: couldn't remove directory %q: %v\n", volumeDir, err)
+			continue
+		}
+
+		parentDir := filepath.Dir(volumeDir)
+		if parentDir == "." || parentDir == "/" {
+			continue
+		}
+
+		if err = os.Remove(parentDir); err != nil {
+			e2e.Logf("WARNING: couldn't remove directory %q: %v\n", parentDir, err)
+			continue
+		}
+	}
+	return nil
+}
+
+// CreateNFSBackedPersistentVolume creates a persistent volume backed by a pod
+// running nfs
+func CreateNFSBackedPersistentVolume(name, capacity, server string) *kapiv1.PersistentVolume {
+	e2e.Logf("Creating persistent volume")
+	return &kapiv1.PersistentVolume{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolume",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"name": name},
+		},
+		Spec: kapiv1.PersistentVolumeSpec{
+			PersistentVolumeSource: kapiv1.PersistentVolumeSource{
+				NFS: &kapiv1.NFSVolumeSource{
+					Server:   server,
+					Path:     "/exports/jenkins-data",
```

Comment:
maybe this path should be an argument too?  or randomly generated?  (right now it seems like you can only call CreateNFSBackedPersistentVolume once per NFS server?)

Reply:
(also doesn't something have to define this export on the NFS server?)

Reply:
(or are the exports hardcoded into your NFS image?)

Reply:
The exports are hard-coded in the image right now, I will make the path more generic (/exports/data) and I can follow up with making it an argument when i inline the Dockerfile

Reply:
sounds good.  for now you might consider at least exporting 10 mounts, just so someone else using the image who needs more than 1 PV can have them.